http/views.go

```diff
@@ -18,9 +18,13 @@ import (
 )
 
 // RobotsTXT return robots.txt valid for complete disallow
-func RobotsTXT(w http.ResponseWriter, r *http.Request) {
+func RobotsTXT(opts *ServerOpts) func(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintln(w, "User-Agent: *")
-	fmt.Fprintln(w, "Disallow: /")
+	if opts.Index {
```

Comment:
is this line working? I don't think you can use return types inside func :)

you should need to return a new function and use the writer

```
func Fetch(opts *ServerOpts) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
...
        }
}
```